src/DeviceInterface.cpp

```diff
@@ -80,7 +80,13 @@ int halide_copy_to_device(void *user_context, struct buffer_t *buf,
 /** Wait for current GPU operations to complete. Calling this explicitly
  * should rarely be necessary, except maybe for profiling. */
 int halide_device_sync(void *user_context, struct buffer_t *buf) {
-    return false;
+    user_assert(user_context == NULL) << "Cannot provide user_context to libHalide.a halide_device_sync\n";
```

Comment:
The functions in DeviceInterface.cpp are the versions that gets called when
jitting. In that case the user_context is hijacked for other purposes. At
least I think that's what's going on - I copied-pasted this assert from the
other functions in the file.

On Thu, Nov 5, 2015 at 10:21 AM, Steven Johnson <notifications@github.com>
wrote:

> In src/DeviceInterface.cpp
> <https://github.com/halide/Halide/pull/961#discussion_r44047807>:
>
> > @@ -80,7 +80,13 @@ int halide_copy_to_device(void *user_context, struct buffer_t *buf,
> >  /** Wait for current GPU operations to complete. Calling this explicitly
> >   * should rarely be necessary, except maybe for profiling. */
> >  int halide_device_sync(void *user_context, struct buffer_t *buf) {
> > -    return false;
> > +    user_assert(user_context == NULL) << "Cannot provide user_context to libHalide.a halide_device_sync\n";
>
> Why is user_context forbidden here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/961/files#r44047807>.
>